clients/src/test/java/org/apache/kafka/clients/consumer/internals/AbstractStickyAssignorTest.java

```diff
@@ -724,6 +725,90 @@ public void testLargeAssignmentAndGroupWithNonEqualSubscription(boolean hasConsu
         assignor.assignPartitions(partitionsPerTopic, subscriptions);
     }
 
+    @Timeout(90)
```

Comment:
Just an opinion here: Other tests also use timeout annotation, but it might be better to use `TestUtils.waitForCondition(()` from other tests because it directly tests the timeout of the `assignPartitions` invocation.  To me, it is a bit more precise about where the tests are failing.  Also, any specific reason we use 90?  I assume the unit is second? I see there's a variation of timeout we are using, from 30->90 😅.

Reply:
Oh, this 90 should be sec, I set 90 sec because in my m1 air this test may run about 25 sec or more time.

I tried ``TestUtils.waitForCondition`` but seems it is not effective and couldn't throw Exception when execution use time more than expected, I'm not familiar with this and couldn't find the reason.

Reply:
Ah no problem.

25s is surprisingly long!